test/Microsoft.ML.Tests/CSharpCodeGen.cs

```diff
@@ -2,24 +2,47 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using Microsoft.ML.Runtime.RunTests;
 using Microsoft.ML.TestFramework;
 using System.IO;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Microsoft.ML.Tests
 {
-    public class CSharpCodeGen : BaseTestClass
+    public class CSharpCodeGen : BaseTestBaseline
     {
         public CSharpCodeGen(ITestOutputHelper output) : base(output)
         {
         }
 
-        [Fact(Skip = "Temporary solution(Windows ONLY) to regenerate codegenerated CSharpAPI.cs")]
-        public void GenerateCSharpAPI()
+        [Fact(Skip = "Execute this test if you want to regenerate CSharpApi file")]
+        public void RegenerateCSharpApi()
         {
-            var cSharpAPIPath = Path.Combine(RootDir, @"src\\Microsoft.ML\\CSharpApi.cs");
-            Runtime.Tools.Maml.Main(new[] { $"? generator=cs{{csFilename={cSharpAPIPath}}}" });
+            var basePath = GetDataPath("../../src/Microsoft.ML/CSharpApi.cs");
+            Runtime.Tools.Maml.Main(new[] { $"? generator=cs{{csFilename={basePath}}}" });
+        }
+
+        [Fact]
+        public void TestGeneratedCSharpAPI()
+        {
+            var dataPath = GetOutputPath("Api.cs");
+            Runtime.Tools.Maml.Main(new[] { $"? generator=cs{{csFilename={dataPath}}}" });
+
+            var basePath = GetDataPath("../../src/Microsoft.ML/CSharpApi.cs");
+            using (StreamReader baseline = OpenReader(basePath))
+            using (StreamReader result = OpenReader(dataPath))
+            {
+                for (; ; )
+                {
+                    string line1 = baseline.ReadLine();
+                    string line2 = result.ReadLine();
+
+                    if (line1 == null && line2 == null)
+                        break;
+                    Assert.Equal(line1, line2);
```

Comment:
I believe you're changing this into a baseline test. Don't baseline tests have a `CheckEquality` utility function, and have to have `Done()` called at the end for an actual failure?